astropy/io/ascii/tests/test_write.py

```diff
@@ -694,24 +694,16 @@ def test_write_overwrite_ascii(format, fast_writer, tmpdir):
     t = table.Table([['Hello', ''], ['', '']], dtype=['S10', 'S10'])
 
     with pytest.raises(OSError) as err:
-        t.write(filename, overwrite=False, format=format,
-                fast_writer=fast_writer)
-    assert str(err.value).endswith('already exists')
-
-    with pytest.warns(
-            AstropyDeprecationWarning,
-            match=r".* Automatically overwriting ASCII files is deprecated. "
-            "Use the argument 'overwrite=True' in the future.") as warning:
         t.write(filename, format=format, fast_writer=fast_writer)
-    assert len(warning) == 1
+    assert str(err.value).endswith("already exists. If you mean to replace "
```

Comment:
While you are updating this, might as well use `pytest.raises(OSError, match=...)` instread of checking `err.value` like this?

Reply:
That can be implemented if it is indeed considered preferable.

Reply:
I prefer `match=` but we can wait for a second opinion. 😸

Reply:
I also prefer `match=`

Reply:
I'll redo the second commit then.